### PR TITLE
Add calendar on events list page

### PR DESCRIPTION
**Pull request by v1k45**
#### Description of changes

Use react-calendar to show events by month
Add events API to get events by month

https://user-images.githubusercontent.com/7877501/120924320-bb320400-c6f0-11eb-9b45-4a0596b86c6e.mp4

### CODE REVIEW

**Comment by johnbxu** [comment]
On some screen sizes the circles look squashed - 
image

**Comment by johnbxu** [comment]
When there are multiple events on the same date, they should all show in the popup: 
image

**Comment by v1k45** [comment]
Thanks for the review. I'll work on a fix for this.

File: events/views.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    try:
+        month = int(request.GET.get('month', ''))
+        year = int(request.GET.get('year', ''))
+    except ValueError:
+        now = timezone.now()
+        month = now.month
+        year = now.year
+
+    events = []
+    queryset = EventPage.objects.live().filter(publishing_date__year=year, publishing_date__month=month)
```
Comment (Parbhat) [inline events/views.py:20]: You can also add the [public](https://docs.wagtail.io/en/stable/reference/pages/queryset_reference.html#wagtail.core.query.PageQuerySet.public) method to the queryset to exclude private pages.

**Comment by johnbxu** [comment]
@v1k45 Thanks for your work on this! You may consider it complete.

File: events/views.py (hunk 1 above)
Comment (v1k45) [reply events/views.py]: Addressed in 98b0189.

**Comment by v1k45** [comment]
Thanks, @johnbxu! I added a commit to address Parbhat's request. It just filters out private pages.

**Comment by v1k45** [comment]
@johnbxu - @Parbhat pointed out the fact that there is a huge changeset on the `package-lock.json` file. The changes were made automatically as a result of `npm install`.

Generally the changeset shouldn't be this large. This could happen if the lockfile isn't regenerated for a while.

Could you confirm if that is OK? I can look into what's causing the this change.

**Comment by johnbxu** [comment]
> @johnbxu - @Parbhat pointed out the fact that there is a huge changeset on the `package-lock.json` file. The changes were made automatically as a result of `npm install`.
> 
> Generally the changeset shouldn't be this large. This could happen if the lockfile isn't regenerated for a while.
> 
> Could you confirm if that is OK? I can look into what's causing the this change.

No problem - that was due to different lockfile versions. I resolved it.

**Review by johnbxu — APPROVED** [review]

**Pull request merged**